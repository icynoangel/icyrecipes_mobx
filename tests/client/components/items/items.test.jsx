import React from 'react';
import ReactDOM from 'react-dom';
import Items from './../../../../src/client/js/components/items/items';
import {shallow} from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

describe('<Items />', function() {
  this.items = [
    {
      itemImage: '/boat.jpg',
      itemTitle: 'Boat',
      itemDescription: 'Lonely boat'
    },
    {
      itemImage: '/rain.jpg',
      itemTitle: 'Rain',
      itemDescription: 'Sunset rain'
    }
  ];

  this.getComponent = items => {
    return <Items.wrappedComponent ItemsStore={{items}} />;
  };

  it('Should render <Items />', () => {
    const wrapper = shallow(this.getComponent(this.items));
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render no Item', () => {
    const itemsList = [];
    const wrapper = shallow(this.getComponent(itemsList));

    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });
});
