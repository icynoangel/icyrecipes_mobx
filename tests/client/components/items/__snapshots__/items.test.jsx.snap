// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items /> Should render <Items /> 1`] = `
<div
  className="items"
>
  <Item
    item={
      Object {
        "itemDescription": "Lonely boat",
        "itemImage": "/boat.jpg",
        "itemTitle": "Boat",
      }
    }
    key="item-Boat"
  />
  <Item
    item={
      Object {
        "itemDescription": "Sunset rain",
        "itemImage": "/rain.jpg",
        "itemTitle": "Rain",
      }
    }
    key="item-Rain"
  />
</div>
`;

exports[`<Items /> Should render no Item 1`] = `
<div
  className="items"
/>
`;
